exercises/linked-list/src/example.h

```diff
@@ -0,0 +1,34 @@
+#ifndef LINKED_LIST_H
+#define LINKED_LIST_H
+
+#include <stdbool.h>
+
+typedef int ll_data_t;
```

Comment:
Since this data type is fixed, is there any reason to expose `list_item` to the outside world? What if we the interface were `push(int)` and `int pop()`?

Reply:
I did have `int` initially but wanted to expose the idea of "variable" type to indicate that the list could really hold anything. 
I tried pushing and popping `ll_data_t` (or `int`) instead of `struct node` to see how it looked, but it makes reaching the empty list state for popping and shifting hard to detect by the caller - this has particular import when destroying the entire list as, unless re-implementing popping in `destroy_list()`, the implementation needs to understand when `shift()` has reached the end of the list - it can't do this if `shift()` only returns an `int`.

Maybe there's another way around this that isn't too complex for an exercise? I'd be happy to adjust to suggestions.

Reply:
BTW, it could be simpler in this regard if the caller took care of node allocation and freeing, but I wanted to ensure the implementation included these concerns... maybe this was wrong and the exercise should concentrate more exclusively on the linked list algorithm?

Reply:
I think that makes sense. I think I'm approaching this from a "what would I do if this was production code" vs. "what's the best way to help someone learn".

Reply:
I struggle with this too! I rewrote the implementation a few times due to previous incantations being what I would prefer in production shoehorned into an exercise... It wasn't pretty. This is partly why I discarded the non-singleton idea early on - I'm still thinking if I could re-implement on this front though.